netcon/mtclient/chat_api.cpp

```diff
@@ -205,7 +205,7 @@ int ConnectToChatServer(const char *serveraddr, int16_t chat_port, char *nicknam
 #ifndef __LINUX__
       if (WSAEWOULDBLOCK == WSAGetLastError())
 #else
-      if (WSAEWOULDBLOCK == ret || 0 == ret)
+      if (WSAEWOULDBLOCK == ret || EINPROGRESS == ret || 0 == ret)
```

Comment:
`WSAEWOULDBLOCK` seems to be a windows-only error code, we shouldn't need to test for it in this block that is only compiled on *NIX

Reply:
It's defined to a similar valid error code on non-Windows, but it's a code not returned by `connect()`. So it is a useless test here. However other such instances in the code base do check for all three error codes on non-Windows so I wanted to keep it similar to existing code.

Reply:
if it's useless then we should remove it

Reply:
Done

Reply:
thanks